hnn_core/mpi_child.py

```diff
@@ -54,20 +54,20 @@ def __exit__(self, type, value, traceback):
             from mpi4py import MPI
             MPI.Finalize()
 
-    def _read_params(self):
-        """Read params broadcasted to all ranks on stdin"""
+    def _read_obj(self):
```

Comment:
Is the goal here to disentangle HNN vocabulary from generic MPI processes? I don't see an immediate use for this outside of passing the `Network` around. Not a big issue though considering the calls to this function are pretty transparent.

Reply:
yes exactly ... if we can make things a bit more general/reusable, we can document and test them independently of the simulations. I don't think I achieved that though :)